interpreter/CMakeLists.txt

```diff
@@ -470,9 +470,18 @@ if (builtin_cling)
     set(CLING_CXXFLAGS " ${LLVM_DEFS} -fno-strict-aliasing -Wwrite-strings -Wno-shadow -Wno-unused-parameter -Wno-deprecated-declarations")
   endif()
 
+  # Requires the linker to resolve the symbol internally and prevents
+  # conflicts when linked with another software using also LLVM like in
+  # the problem reported for Julia in
+  # https://github.com/JuliaHEP/ROOT.jl/issues/17#issuecomment-882719292
+  # Only needed for Linux: Mac uses linker namespaces and  Windows explicit export/import
+  if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
```

Comment:
I'd appreciate to have this close line 135 `ROOT_ADD_CXX_FLAG(CMAKE_CXX_FLAGS "-fno-semantic-interposition")` as these seem related. Do we still need `-fno-semantic-interposition` or is `-Wl,-Bsymbolic` a superior replacement?

Reply:
For our needs, -Wl,-Bsymbolic would be sufficient, but the two flags are not incompatible. Furthermore, the `-fno-semantic-interposition` could lead to more performant code as it provides information to the compiler (the first is only known to the linker). I expect that thanks to this flags, the compiler can decide to inline functions, not marked as "inline" in the code, it would not be able to do otherwise, with the caveat I am not expert in the compilation optimization matter.

So I would keep both. I agree to group the two lines in the same place. For the architecture condition, do you prefer `${CMAKE_SYSTEM_NAME} MATCHES "Linux"` or `if (NOT MSVC AND NOT APPLE)` ?

Reply:
I prefer `if (NOT MSVC AND NOT APPLE)`: that exposes it to more platforms, and if things fail on one of those we can suppress these platforms as needed. Where to do that will be a simple grep based on "I am your compiler and I don't understand -fno-semantic-interposition". Thanks!